packages/core/TextSearch/TextSearchManager.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/*  text-searching controller */
+import BaseResult from './BaseResults'
+import { AnyConfigurationModel } from '../configuration/configurationSchema'
+
+import PluginManager from '../PluginManager'
+import QuickLRU from '../util/QuickLRU'
+import { searchType, BaseTextSearchAdapter } from '../data_adapters/BaseAdapter'
+import { readConfObject } from '../configuration'
+import TextSearchAdapterType from '../pluggableElementTypes/TextSearchAdapterType'
+
+export interface BaseArgs {
+  searchType: searchType
+  queryString: string
+  signal?: AbortSignal
+  limit?: number
+  pageNumber?: number
+}
+
+export interface Scope {
+  aggregate: boolean
+  assemblyName: string
+  openedTracks?: Array<string>
```

Comment:
rename to just `tracks`. And if there are tracks in a search scope, the search should 100% be limited to only results from those tracks.

The idea of a track being "open" is outside the realm of what the text search manager knows about, it just wants somebody to tell it what parts of the universe of data it should search.

Reply:
Could you elaborate on the first part? Are we thinking that if there are tracks in the search scope, the results thrown should be results coming only from  track specific text search adapters and no aggregate text search adapters(if those tracks have a track specific text search adapter configured) or are we displaying only results that come from these tracks (aggregate adapters could throw a result that comes from covers that track)?